heimctl/cmd/retention.go

```diff
@@ -66,5 +66,9 @@ func (cmd *retentionCmd) run(ctx scope.Context, args []string) error {
 	ctx.WaitGroup().Add(1)
 	retention.ExpiredScanLoop(ctx, c, b, cmd.interval)
```

Comment:
The lack of a "go" here means the DeleteScanLoop launch below is never reached. Either rearrange things so that the last module is started without a new goroutine, or launch them all in goroutines and use ctx.WaitGroup().Wait() before returning ctx.Err().